rustls/src/msgs/message.rs

```diff
@@ -459,6 +459,149 @@ impl<'a> BorrowedPlainMessage<'a> {
     }
 }
 
+#[derive(Debug, Clone)]
+/// A collection of borrowed plaintext slices
+pub enum BorrowedPlainPayload<'a> {
+    Empty,
+    /// A single byte slice. This allows only one redirection instead of several.
+    Single(&'a [u8]),
+    /// A collection of chunks (byte slices)
+    /// and inclusive-exclusive cursors to single out a slice of bytes.
+    /// [BorrowedPayload] assumes three invariants:
+    /// - at least two chunks
+    /// - a start cursor pointing into the first chunk
+    /// - an end cursor pointing into the last one
+    Multiple {
+        chunks: &'a [&'a [u8]],
+        start: usize,
+        end: usize,
+    },
+}
+
+impl<'a> BorrowedPlainPayload<'a> {
+    /// Create a payload with a single byte slice
+    pub fn new_single(payload: &'a [u8]) -> Self {
+        Self::Single(payload)
+    }
```

Comment:
This gets called fairly frequently through the change set. I wonder if it would make sense to impl `From<&'a [u8]>` for this purpose and leave just `new` for the from chunks constructor. WDYT?